src/workerd/server/server-test.c++

```diff
@@ -2103,6 +2103,98 @@ KJ_TEST("Server: disk service allow dotfiles") {
   KJ_EXPECT(test.root->openFile(kj::Path({"secret"}))->readAllText() == "this is super-secret");
 }
 
+
+KJ_TEST("Cache: If no cache service is defined, access to the cache API should error") {
+  TestServer test(singleWorker(R"((
+    compatibilityDate = "2022-08-17",
+    modules = [
+      ( name = "test.js",
+        esModule =
+          `export default {
+          `  async fetch(request) {
+          `    try {
+          `      return new Response(await caches.default.match(request))
+          `    } catch (e) {return new Response(e.message)}
+          `
+          `  }
+          `}
+      )
+    ]
+  ))"_kj));
+
+
+  test.start();
+  auto conn = test.connect("test-addr");
+  conn.httpGet200("/",
+      "No Cache was configured");
+
+}
+
+
+KJ_TEST("Cache: cached response") {
+  TestServer test(R"((
+    services = [
+      ( name = "hello",
+        worker = (
+          cacheApiOutbound = "cache-outbound",
+          compatibilityDate = "2022-08-17",
+          modules = [
+            ( name = "main.js",
+              esModule =
+                `export default {
+                `  async fetch(request, env, ctx) {
+                `    const cache = caches.default;
+                `    let response = await cache.match(request);
+                `    return response ?? new Response('not cached');
+                `  }
+                `}
+            )
+          ]
+        )
+      ),
+      ( name = "cache-outbound", external = "cache-host" ),
+    ],
+    sockets = [
+      ( name = "main",
+        address = "test-addr",
+        service = "hello"
+      )
+    ]
+  ))"_kj);
+
+  test.start();
+  auto conn = test.connect("test-addr");
+  conn.sendHttpGet("/");
+
+  {
+    auto subreq = test.receiveSubrequest("cache-host");
+    subreq.recv(R"(
+      GET / HTTP/1.1
+      Host: foo
+      Cache-Control: only-if-cached
+
+    )"_blockquote);
+    subreq.send(R"(
+      HTTP/1.1 200 OK
+      CF-Cache-Status: HIT
+      Content-Size: 6
+
+      cached)"_blockquote);
+  }
+
+  conn.recv(R"(
+    HTTP/1.1 200 OK
+    Transfer-Encoding: chunked
+    CF-Cache-Status: HIT
+    Content-Size: 6
+
+    6
+    cached
+    0
+
+  )"_blockquote);
+
+}
```

Comment:
Let's make sure to test `caches.open()` to make sure the name header comes through as expected.